### DISCUSSION TITLE
How to check if coordinate[] can form a valid ring

### DISCUSSION THREAD

**Discussion by hdahmani1** [NetTopologySuite/NetTopologySuite#661] [Q&A] [answered]
Hi, 
I have a situation where i have to calculate the centroid of an array of coordinate. So i have to create LinearRing to have the result. But, before, i have to check that the given coordinates can form a valid ring... so it's a spiral problem !
Any ideas ?
Thank you

**Comment by DGuidi** [comment, marked as answer]
Why not build a linear ring directly? If something wrong happens, you receive an error, so the coordinates are not closed (or you have other validity issues)

**Reply by hdahmani1** [reply]
I solved the problem by using a try catch statement. Thank you DGuidi!
